File: 2018-19/unit6/mario/more/mario.py
'''
    Tahmidul Alam
    CS50AP
    Mario (more)
    mario.py
'''

from cs50 import get_int


def main():
    # User validation
    while True:
        print("Height: ", end="")
        height = get_int()
        if height >= 0 and height <= 23:
            break

    # Standard mario.py Start
    for i in range(height):
        for j in range(height - i - 1):
            print(" ", end="")
        for k in range(i + 1):
            print("#", end="")
    # Standard mario.py End
    # Start the right pyramid (put a space after the first and
    # continue)
        print("  ", end="")
    # Finally, print the right pyramid along with the left pyramid
        for m in range(i + 1):
            print("#", end="")
    # Print a new line
        print()


# Main Function
if __name__ == "__main__":
    main()

File: 2017-18/chapter1/mario/mario.c
/**
 * Tahmidul Alam
 * mario.c
 * Ms. Tanzosh
 *
*/
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    int half_pyramid_height;
    half_pyramid_height = 0;
    do
    {
        half_pyramid_height = get_int("Height: ");
    }
    while (half_pyramid_height < 0 || half_pyramid_height > 23);

    for (int i = 0; i < half_pyramid_height; i++)
    {
        for (int spacing = 0; spacing < (half_pyramid_height - i) - 1; spacing++)
        {
            printf(" ");
        }

        for (int pyramid = 0; pyramid < i + 2; pyramid++)
        {
            printf("#");
        }
        printf("\n");
    }
    return 0;
}